src/providers/swift.rs

```diff
@@ -0,0 +1,147 @@
+use super::Provider;
+use crate::nixpacks::{
+    app::App,
+    environment::Environment,
+    nix::pkg::Pkg,
+    phase::{BuildPhase, InstallPhase, SetupPhase, StartPhase},
+};
+use anyhow::Result;
+use indoc::formatdoc;
+
+static LATEST_SWIFT: &str = "5.6.1";
+
+pub struct SwiftProvider {}
+
+impl Provider for SwiftProvider {
+    fn name(&self) -> &str {
+        "swift"
+    }
+
+    fn detect(&self, app: &App, _env: &Environment) -> Result<bool> {
+        Ok(app.includes_file("Package.swift")
+            || (app.includes_file("Package.swift") && app.includes_file("Package.resolved")))
+    }
+
+    fn setup(&self, _app: &App, _env: &Environment) -> Result<Option<SetupPhase>> {
+        let pkgs = vec![Pkg::new("python38"), Pkg::new("wget")];
+
+        Ok(Some(SetupPhase::new(pkgs)))
+    }
+
+    fn install(&self, app: &App, _env: &Environment) -> Result<Option<InstallPhase>> {
+        let version = SwiftProvider::get_swift_version(&app)?;
+        let (download_url, name) = SwiftProvider::make_download_url(version);
+
+        // https://forums.swift.org/t/which-clang-package-should-we-install/20542/14
+        let install_cmd = formatdoc! {"
+        sudo apt-get update && \
+        sudo apt-get install -y build-essential clang libsqlite3-0 libncurses6 libcurl4 libxml2 libatomic1 libedit2 libsqlite3-0 libcurl4 libxml2 libbsd0 libc6-dev && \
+        wget -q {download_url} && \
+        tar -xf {name}.tar.gz && \
+        sudo mv {name} /usr/share/swift
+        ",
+        name=name,
```

Comment:
I'm surprised cargo fmt didn't pick this up, but could it be name = name? And same for the other ones.

Reply:
done